agenda.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+use Sabre\VObject;
+
+class NotImplementedException extends BadMethodCallException {}
+
+require __DIR__ . '/vendor/autoload.php';
+
+class Agenda {
```

Comment:
IIUC this class may be called in a concurrent context (I guess Slack can call our index.php several times at almost the same time if several users uses this bot) and it may be tricky because it uses the filesystem which is a shared resource.
However, it's perhaps not a big deal to not deal with concurrency the worst thing that could happen is that we would do useless queries to the calendar but no state (either in ram in the filesystem) should be corrupted.

Does this analysis sounds correct to you? Do you have thoughts on this?

Reply:
This could be an issue indeed. I have added file locks in 58a045e and 9fb9099 to prevent concurrent read/write.